src/focus-service.ts

```diff
@@ -111,75 +58,21 @@ export class FocusService {
    * This can be Infinity to disable the animation, or null to disable scrolling.
    */
   public scrollSpeed: number | null = 1000;
-  public focusRoot: HTMLElement = defaultFocusRoot;
-  // The currently selected element.
-  public selected: HTMLElement | null = null;
-  // Focus root, the service operates below here.
-  private root: HTMLElement | null = null;
   // Subscription to focus update events.
   private registrySubscription?: Subscription;
 
   private prevElement: HTMLElement | undefined;
   // The client bounding rect when we first selected the element, cached
   // so that we can reuse it if the element gets detached.
   private referenceRect: ClientRect = defaultRect;
-  private focusStack: IFocusState[] = [];
   private focusByRegistry = new FocusByRegistry();
 
-  constructor(private registry: StateContainer) {}
-
-  public trapFocus(newRootElem: HTMLElement) {
-    this.focusStack.push({
-      focusedElem: this.selected,
-      root: this.focusRoot,
-    });
-    this.focusRoot = newRootElem;
-  }
-
-  public releaseFocus(releaseElem?: HTMLElement, scrollSpeed: number | null = this.scrollSpeed) {
-    if (releaseElem) {
-      if (releaseElem === this.focusRoot) {
-        this.releaseFocus(undefined, scrollSpeed);
-      }
-      return;
-    }
-
-    const lastFocusState = this.focusStack.pop();
-    if (lastFocusState && lastFocusState.focusedElem) {
-      this.focusRoot = lastFocusState.root;
-      this.selectNode(lastFocusState.focusedElem, scrollSpeed);
-    } else {
-      throw new Error(
-        'No more focus traps to release. Make sure you call trapFocus before using releaseFocus',
-      );
-    }
-  }
-
-  /**
-   * Useful for resetting all focus traps e.g. on page navigation
-   */
-  public clearAllTraps() {
-    this.focusStack.length = 0;
-    this.focusRoot = defaultFocusRoot;
+  private get selected() {
+    return document.activeElement as HTMLElement;
```

Comment:
we wanna keep `document.activeElement` separate from `focus.selected` to support showing the elements as focused without actually focusing them. This is necessary for smooth scrolling to work

Reply:
@saadataz it looks like there's a (new?) [preventScroll option](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus) of the .focus() method. That seems to fill the use case of smooth scroll.